polars/polars-lazy/src/physical_plan/executors/cache.rs

```diff
@@ -8,22 +8,35 @@ pub struct CacheExec {
 
 impl Executor for CacheExec {
     fn execute(&mut self, state: &mut ExecutionState) -> PolarsResult<DataFrame> {
-        if self.count > 0 {
```

Comment:
I wanted to double check and make sure I'm understanding the purpose of `self.count`. It seems to be used for cache expiration? When `count` is zero, the execution should always run? I keep seeing `count` initialized to `UINT_MAX`, so I'm not sure where the use case of this ever expiring happens.